crates/transaction-pool/src/traits.rs

```diff
@@ -969,6 +972,12 @@ pub struct EthPooledTransaction {
     pub(crate) blob_sidecar: EthBlobTransactionSidecar,
 }
 
+impl IntoRecoveredTransaction for EthPooledTransaction {
+    fn to_recovered_transaction(&self) -> TransactionSignedEcRecovered {
+        self.transaction.clone()
+    }
+}
```

Comment:
I don't think we use this on main, so idk why we need this here

Reply:
agreed. 

(note: name of the trait should be `ToRecoveredTransaction`, ref https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv)

Reply:
Just to clarify @mattsse, did you want me to remove all traces of `IntoRecoveredTransaction` from the codebase? Or just remove this impl and the new trait bounds? Either one of these is possible.